browser/ui/webui/brave_rewards_ui.cc

```diff
@@ -1306,82 +1316,81 @@ void RewardsDOMHandler::FetchBalance(const base::ListValue* args) {
-      transaction.SetStringKey("date", std::to_string(tx.date));
-      transaction.SetIntKey("category", tx.category);
-      transactions.GetList().push_back(std::move(transaction));
-    }
-    monthly.SetKey("transactions", base::Value(std::move(transactions)));
-
-    base::Value report(base::Value::Type::DICTIONARY);
-    report.SetStringKey("opening", monthly_statement->report.opening_balance);
-    report.SetStringKey("closing", monthly_statement->report.closing_balance);
-    report.SetStringKey("grant", monthly_statement->report.grants);
-    report.SetStringKey("ads", monthly_statement->report.earning_from_ads);
-    report.SetStringKey("contribute",
-        monthly_statement->report.auto_contribute);
-    report.SetStringKey("donation",
-        monthly_statement->report.recurring_donation);
-    report.SetStringKey("tips", monthly_statement->report.one_time_donation);
-    report.SetStringKey("total", monthly_statement->report.total);
-    monthly.SetKey("balanceReport", base::Value(std::move(report)));
-
-    base::Value months(base::Value::Type::LIST);
-    for (const std::string& month : monthly_statement->months_available) {
-      months.GetList().push_back(base::Value(month));
-    }
-    monthly.SetKey("monthsAvailable", base::Value(std::move(months)));
-
-    monthly.SetStringKey("reconcileStamp",
-        std::to_string(monthly_statement->reconcile_stamp));
-
-    web_ui()->CallJavascriptFunctionUnsafe(
-         "brave_rewards.onGetMonthlyStatements",
-         std::move(monthly));
+void RewardsDOMHandler::OnGetOneTimeTipsStatements(
+    const std::string& webui_callback_id,
+    std::unique_ptr<brave_rewards::ContributionInfoList> list) {
+  LOG(ERROR) << "+++++++++IN ONGETONETIMETIPSSTATEMENTS";
+  AllowJavascript();
+  base::Value one_time_tips(base::Value::Type::LIST);
+  for (const auto& item : *list) {
+    base::Value tip(base::Value::Type::DICTIONARY);
+    LOG(ERROR) << "==========PUB KEY: " << item.publisher_key;
+    tip.SetStringKey("publisherKey", item.publisher_key);
+    tip.SetStringKey("probi", item.probi);
+    tip.SetStringKey("date", std::to_string(item.date));
+    tip.SetIntKey("category", item.category);
+    one_time_tips.GetList().push_back(std::move(tip));
   }
+
+  ResolveJavascriptCallback(
+      base::Value(webui_callback_id),
+      std::move(one_time_tips));
 }
 
-void RewardsDOMHandler::GetMonthlyStatements(const base::ListValue* args) {
-  CHECK_EQ(2U, args->GetSize());
+void RewardsDOMHandler::GetOneTimeTipsStatements(
+    const base::ListValue* args) {
+  LOG(ERROR) << "=========IN GETONETIMETIPSSTATEMENTS";
+  CHECK_EQ(3U, args->GetSize());
+  std::string webui_callback_id = args->GetList()[0].GetString();
   if (!rewards_service_) {
     return;
   }
-  int32_t month = args->GetList()[0].GetInt();
-  int32_t year = args->GetList()[1].GetInt();
-  rewards_service_->GetMonthlyStatements(
+  int32_t month = args->GetList()[1].GetInt();
```

Comment:
I know this was already like this, but we should avoid `args->GetList()[]` in favor of `args->GetInteger(index, &month)`

Reply:
oh... actually I guess not

Reply:
```// DEPRECATED, use GetList()::operator[]::GetInt() instead.``` huh, wouldn't have thought that